### ISSUE TITLE
Go parameters start with struct name for methods

### ISSUE THREAD

**Issue by mattiasberlin** [RobertOstermann/vscode-inline-parameters#3]
Thanks for adding Go support! By chance I found the extension a day after you added support for it.

It works great if you call functions that are not methods. But if it's a method then the parameters start with the struct name, which it should not do.

Here's a repro:
```go
package main

type A struct {
	s string
}

func (a A) addSuffix(suffix string) string {
	return a.s + suffix
}

func abc(a string) string {
	return a
}

func main() {
	abc("test") // Works as expected.

	a := A{}
	a.addSuffix("test") // Inlines "A:" (the struct name) when it should be "suffix:".
}
```

**Comment by RobertOstermann** [comment]
@mattiab-axis Thanks for bringing this to my attention and providing an example for me to use. I have never used Golang before so that is extremely helpful. I have released version 1.1.0 which should address this problem.

**Comment by mattiasberlin** [comment]
Thank you, methods seems to work like they should now!

I noticed another oddity, which seems to occur when you call a function from another package which takes an interface as a parameter. I don't know if you want to continue in this issue or create another one, but I'll give a repro here. Unfortunately since it requires two packages it needs a little more setup.

anotherpackage/anotherpackage.go
```go
package anotherpackage

type Struct struct{}

func (a Struct) Method(s string) string {
	return s
}

func Function(s string) string {
	return s
}

type Interface interface {
	Method(string) string
}

func FunctionWithInterfaceParameter(a Interface, s string) string {
	return a.Method(s)
}
```

example.go
```go
package main

import "example/anotherpackage"

func main() {
	anotherpackage.Function("test") // Correct.

	a := anotherpackage.Struct{}
	a.Method("test") // Correct.

	anotherpackage.FunctionWithInterfaceParameter(a, "test") // First param is correct. Second should be "s:" but is "Interface:" (the type name of the first param).
}
```
Then either run `go mod init example` or create a `go.mod` with this content (if you're running Go 1.18 that is):
```
module example

go 1.18
```

**Comment by RobertOstermann** [comment]
ok, I looked into that problem. Looks like it is a bug because I messed up the regex to parse the variadic parameters. I have published version 1.1.1 that should fix that bug.

**Comment by mattiasberlin** [comment]
Awesome, thanks a lot! I'll close the task.
